packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -16,6 +16,9 @@ import 'viewport_offset.dart';
 // CORE TYPES FOR SLIVERS
 // The RenderSliver base class and its helper types.
 
+/// Called to get the item extent by the index of item.
+typedef ItemExtentGetter = double Function(int index);
```

Comment:
What do you think of passing an index plus a little bit more information about the viewport?

My original thought for this was to see what folks thought of the TableSpanExtent API in the TableView (https://github.com/flutter/packages/pull/4536) and maybe move that into the framework.
This section of the design document talks about it (links direct to that section): https://docs.google.com/document/d/15ecTZE1g3WeswLGFWrnEgMP6SyL6jDRdxOgPsczOcV0/edit?resourcekey=0-yNd_qFhiPjz6z2TgezWc0A#heading=h.a8uex5ucyiso

It defines multiple ways for developers to set an extent, based on the viewport, or the preceding scroll extent, or as the result of an operation, etc. 

I figured if it worked well for folks (we did already test it in a usability study), we could use it here as well, making it a super class like `ChildScrollExtent` or some such - names are hard. 😆

Reply:
```What do you think of passing an index plus a little bit more information about the viewport?```

What do you think of exposing the `SliverConstraints` directly or just some of the key properity of viewport?

Reply:
Hmmm, that's a good idea! It won't communicate the information about how much the SliverList has laid out already though, since the SliverConstraints relate to the whole sliver and not its contents. What parts of SliverConstraints would be relevant to the user when building an item of the list? Definitely the viewportExtent I think. Maybe we can wrap those parts up in an object with additional info like how much extent has been laid out.

Reply:
I have wraped some layout relation properity into `SliverLayoutDimensions` and exposed it to the callback.

Will finish the test cases ASAP.